framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -951,19 +967,28 @@ public JavaExpression visit(BinaryExpr expr, Void aVoid) {
       TypeMirror leftType = leftJe.getType();
       TypeMirror rightType = rightJe.getType();
       TypeMirror type;
+      Supplier<ParseRuntimeException> exception =
+          () ->
+              new ParseRuntimeException(
+                  constructJavaExpressionParseError(
+                      expr.toString(),
+                      String.format("inconsistent types %s %s for %s", leftType, rightType, expr)));
       // isSubtype() first does the cheaper test isSameType(), so no need to do it here.
-      if (types.isSubtype(leftType, rightType)) {
+      if (COMPARISON_OPERATORS.contains(expr.getOperator())) {
+        if (types.isSubtype(leftType, rightType) || types.isSubtype(rightType, leftType)) {
```

Comment:
I suggest this should be one conjunction, as [below](https://github.com/typetools/checker-framework/pull/6940/files#diff-a773bceb2a0a5e95d55bd84fdbfbc546d38dd66b10d9f1ecc9c821ceab671e3eR987-R988) for the PLUS operator.  Then there is no need for a nested `if` statement nor its `else` clause.